src/com/owncloud/android/ui/activity/LogHistoryActivity.java

```diff
@@ -173,76 +179,83 @@ private void sendMail() {
             startActivity(intent);
         } catch (ActivityNotFoundException e) {
             Snackbar snackbar = Snackbar.make(
-                findViewById(android.R.id.content),
-                R.string.log_send_no_mail_app,
-                Snackbar.LENGTH_LONG
+                    findViewById(android.R.id.content),
+                    R.string.log_send_no_mail_app,
+                    Snackbar.LENGTH_LONG
             );
             snackbar.show();
             Log_OC.i(TAG, "Could not find app for sending log history.");
         }
-
     }
 
     /**
      *
      * Class for loading the log data async
      *
      */
-    private class LoadingLogTask extends AsyncTask<String, Void, String> {
-        private final WeakReference<TextView> textViewReference;
-
-        public LoadingLogTask(TextView logTV){
-            // Use of a WeakReference to ensure the TextView can be garbage collected
-            textViewReference  = new WeakReference<TextView>(logTV);
-        }
+    private class LoadingLogTask extends AsyncTask<String, Void, ArrayList<Log>> {
 
-        protected String doInBackground(String... args) {
+        protected ArrayList<Log> doInBackground(String... args) {
             return readLogFile();
         }
 
-        protected void onPostExecute(String result) {
-            if (textViewReference != null && result != null) {
-                final TextView logTV = textViewReference.get();
-                if (logTV != null) {
-                    mLogText = result;
-                    logTV.setText(mLogText);
-                    dismissLoadingDialog();
-                }
+        protected void onPostExecute(ArrayList<Log> result) {
+            if (result != null) {
+                mLogListAdapter = new LogListAdapter(result);
+                mLogsRecycler.setAdapter(mLogListAdapter);
+                dismissLoadingDialog();
             }
         }
 
         /**
          * Read and show log file info
          */
-        private String readLogFile() {
-
+        private ArrayList<Log> readLogFile() {
```

Comment:
Please chop this methode down in order to make it better readable.

Suggestion:
- `readLogFiles()` which reads all log files
- `readLogFile()` which only reads one file
- `readLineFromLog()` which only reads one line

Reply:
Done